controllers/configuration_controller.go

```diff
@@ -187,6 +196,26 @@ func (r *ConfigurationReconciler) shouldReconcile(ctx context.Context, cv *v1alp
 		return true, nil
 	}
 
+	// if source is a snapshot, check on its status
```

Comment:
We also need to check the config input for the configuration controller. This could be a snapshot too. Then there is also the strategic merge source which is a flux source and that needs to be checked.